mdata/ifaces.go

```diff
@@ -12,6 +13,12 @@ type Metrics interface {
 
 type Metric interface {
 	Add(ts uint32, val float64)
-	Get(from, to uint32) (uint32, []chunk.Iter)
-	GetAggregated(consolidator consolidation.Consolidator, aggSpan, from, to uint32) (uint32, []chunk.Iter)
+	Get(from, to uint32) MetricResult
+	GetAggregated(consolidator consolidation.Consolidator, aggSpan, from, to uint32) MetricResult
+}
+
+type MetricResult struct {
```

Comment:
also I think this warrents some more documentation, it's not really clear what "Raw" is for example and how it relates to Iters. also if the slice of iters is called iters, why is the slice of points called raw and not points?